scripts/edit-task.js

```diff
@@ -0,0 +1,57 @@
+import { CLICK_EVENT, HIDE_CLASS } from "/scripts/constants.js";
+
+function onTaskEdit(
+  doneButton,
+  editButton,
+  deleteButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  toolbar,
+  tasks,
+  id
+) {
+  function onEditButtonClick() {
+    const prevContent = inputElement.innerText;
+    inputElement.contentEditable = true;
+    inputElement.focus();
+    [editButton, deleteButton].forEach((e) => e.classList.add(HIDE_CLASS));
+    [revertButton, saveButton].forEach((e) => e.classList.remove(HIDE_CLASS));
+
+    function commonEvent() {
+      inputElement.contentEditable = false;
+      saveButton.classList.add(HIDE_CLASS);
+      revertButton.classList.add(HIDE_CLASS);
+      deleteButton.classList.remove(HIDE_CLASS);
+    }
```

Comment:
let's refactor the functionality of show/hide'ing buttons in a separate function and use that properly. this will reduce clutter in the handler.

```
function toggleButtons(show, ...buttons) {
    buttons.forEach((e) => {
        if (show) {
            e.classList.remove(HIDE_CLASS);
        } else {
            e.classList.add(HIDE_CLASS);
        }
    });
}
```

```suggestion
   ...

    function commonEvent() {
      inputElement.contentEditable = false;
      ...
      deleteButton.classList.remove(HIDE_CLASS);
    }
    
```